src/sagemaker_containers/modules.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License'). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the 'license' file accompanying this file. This file is
+# distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+import importlib
+import logging
+import os
+import shlex
+import shutil
+import subprocess
+import sys
+import tarfile
+import tempfile
+import traceback
+
+import boto3
+from six.moves.urllib.parse import urlparse
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_MODULE_NAME = 'default_user_module_name'
+
+
+def download(url, dst):
```

Comment:
Maybe "s3_download" since this is s3 specific?

Reply:
sure.

Reply:
Actually, my thought process is the following:
download for now is only S3 and throws exception otherwise.
later, if we are downloading for git for example, we can rename download to s3_download, create a clone function and create a download functions that cover both cases. What do you think about this approach?

Reply:
I think even for now, with only having one function, we should give it a name which is descriptive enough. I get that in general we want short names since its pythonic, but I think the three characters here are worth it.